index.js

```diff
@@ -4,30 +4,41 @@ function emptyTarget(val) {
 	return Array.isArray(val) ? [] : {}
 }
 
-function cloneUnlessOtherwiseSpecified(value, optionsArgument) {
+function cloneUnlessOtherwiseSpecified(value, optionsArgument, key) {
 	var clone = !optionsArgument || optionsArgument.clone !== false
 
-	return (clone && isMergeableObject(value))
+	return (clone && _isMergeableObject(value, optionsArgument, key))
 		? deepmerge(emptyTarget(value), value, optionsArgument)
 		: value
 }
 
+function _isMergeableObject(value, optionsArgument, key) {
+	let ret;
+	if (optionsArgument && optionsArgument.isMergeableObject) {
+		ret = optionsArgument.isMergeableObject(value, isMergeableObject, optionsArgument, key)
+	}
+	if (ret === null || typeof ret === 'undefined') {
+		ret = isMergeableObject(value)
```

Comment:
This seems like a huge issue - we shouldn't fall back to the default function.

This function seems like it should just be something like

```js
function isMergeableObject(value, optionsArgument) {
	const testFunction = (optionsArgument && optionsArgument.isMergeableObject) 
		? optionsArgument.isMergeableObject
		: defaultIsMergeableObject

	return testFunction(value, defaultIsMergeableObject)
}
```